src/amo/components/AddonTitle/index.js

```diff
@@ -25,6 +26,7 @@ type InternalProps = {|
 |};
 
 export const AddonTitleBase = ({
+  ComponentTag = 'h1',
```

Comment:
If you go for `as` or `component`, you can then alias the prop to `Component` so that it can be instantiated by React:

```suggestion
  as: Component = 'h1',
```

Reply:
@willdurand, I've updated using "as" but I am still wondering if that's super clear. Do you think a "tag" or "wrapperTag" prop might be more clear? 

set up in this same way tho:

`tag: Component = 'h1',`